docs/guides/development/nodejs/nodejs-twitter-bot/index.md

```diff
@@ -0,0 +1,294 @@
+
+Up next, we’re going to install all the npm packages we need, so enter the following into your terminal in VSCode.
+
+```
+"Npm init -y"
+
+"Npm install node"
+
+"Npm install node-schedule"
+
+"Npm install twit"
+
+"Npm install dotenv"
+```
+</br>
+This installs all the dependencies we’ll need to be able to have a functional Twitter bot that can read the API, post on a schedule, use the `.env` file, and use `node.js`. We have one more thing to adjust, and that’s to create some scripts inside the `package.json` file code block. Open it and add the following to the scripts section.
+
+```javascript
+	"test": "index.js",
+	      "start": "node ./index.js",
+	      "develop": "NODE_ENV=develop node index.js"
+```
+</br>
+That `index.js` file doesn’t exist yet, so go ahead and create it and name it whatever you like. I’ll be calling mine `snes.js`. Go ahead and make a `config.js` at the same time and input the following into it.
+
+```javascript
+	module.exports = {
+	    consumer_key: process.env.consumer_key,
+	    consumer_secret: process.env.consumer_secret,
+	    access_token: process.env.access_token,
+	    access_token_secret: process.env.access_token_secret,
+	};
+```
+
+After all this, open up `index.js`, or whatever you've named it, and get ready to type some javascript
+
+## Code It Up.
+
+```javascript
+	console.log("SNES Soundtracks booting up");
+
+	//making sure npm run develop works
+	if (process.env.NODE_ENV === "develop") {
+	    require("dotenv").config();
+	};
+
+	//rules for node-schedule
+	var schedule = require("node-schedule");
+	var rule = new schedule.RecurrenceRule();
+	  rule.dayOfWeek = 1,
+	  rule.hour = 10;
+	  rule.tz = "Etc/GMT+4";
+
+	//array to pull soundtracks from
+	var soundtrackArray = [ "an array of youtube video URLs"];
+	var soundtrackArrayLength = soundtrackArray.length;
+	var soundtrackArrayElement = Math.floor(Math.random() * soundtrackArrayLength);
+
+```
+</br>
+
+At the beginning, I log a start up message to the console just so I know it’s running. Next is an if statement to use the dotenv when the node environment is ‘develop’, which is handled in the scripts of the json file from earlier. We set up a few variables for the node-schedule so the bot can tweet on a set day and time. I pulled this directly from the docs for node-schedule. Basically, it will tweet every Monday at 10 am Eastern Daylight Savings Time. Finally, I set up an array for the bot to pull from at random with the last three lines of this section. I removed the URLs for the tutorial.
+
+```javascript
+	// Create a Twitter object to connect to Twitter API
+	var Twit = require('twit');
+
+	// Pulling keys from another file
+	var config = require('./config.js');
+	// Making a Twit object for connection to the API
+	var T = new Twit(config);
+
+	// Setting up a user stream
+	var stream = T.stream('statuses/filter', { track: '@SnesSoundtracks' });
```

Comment:
I moved the `@SnesSoundtracks` string into a new variable at the top of the code, and then referenced that here, and where it appears in the next section of code.

Did this because the reader of the guide will need to pick a different Twitter username for their bot, and refactoring this into a variable means they only need to replace it once when they copy/paste the code snippets.

I considered moving this string into the `.env` file, but I didn't want to make too many changes